regparser/grammar/interpretation_headers.py

```diff
@@ -25,10 +25,16 @@
     + unified.depth1_p
 )
 
-
+# This matches an appendix name in an appendix header. Here we'll match
+# something with a dash in the appendix name (i.e. AA-1) but we'll
+# remove the dash. The effect of this is that, for label purposes only,
+# the appendix becomes known as 'AA1', and therefore we don't have weird
+# label collisions with a node labeled '1' underneath the appendix.
 appendix = (
     atomic.appendix_marker.copy().leaveWhitespace()
-    + (unified.appendix_with_section | atomic.appendix )
+    + Regex(r"[A-Z]+-?[0-9]*\b").setResultsName("appendix").setParseAction(
```

Comment:
Well, no, because `atomic.appendix` is used more often in more varied cases, including in resolving labels later on.

"AA-1" should generally resolve to something like `["1111", "AA", "1"]` rather than `["1111", "AA-1"]`, which is what happens if this replaces `atomic.appendix`.

That's another reason why the action here is to strip out the dash, so that, for the purpose of labels, it becomes "AA1".